pkg/registry/generic/rest/proxy.go

```diff
@@ -202,7 +206,51 @@ func (h *UpgradeAwareProxyHandler) dialURL() (net.Conn, error) {
 
 		return tlsConn, nil
 	default:
-		return nil, fmt.Errorf("Unknown scheme: %s", h.Location.Scheme)
+		return nil, fmt.Errorf("unknown scheme: %s", h.Location.Scheme)
+	}
+}
+
+func dialURLWithDialer(httpTransport *http.Transport, location *url.URL) (net.Conn, error) {
```

Comment:
Missing future fixes due to skew is a fair concern. Combined new logic into dialURL, PTAL